src/module/service.cpp

```diff
@@ -129,6 +129,35 @@ ::grpc::Status ModuleService_::ReconfigureResource(
     return grpc::Status();
 };
 
+::grpc::Status ModuleService_::ValidateConfig(
+    ::grpc::ServerContext* context,
+    const ::viam::module::v1::ValidateConfigRequest* request,
+    ::viam::module::v1::ValidateConfigResponse* response) {
+    viam::app::v1::ComponentConfig proto = request->config();
+    Resource cfg = Resource::from_proto(proto);
+
+    std::shared_ptr<ModelRegistration> reg = Registry::lookup_resource(cfg.api, cfg.model);
+    if (reg == nullptr) {
+        return grpc::Status(grpc::UNKNOWN,
+                            "unable to validate resource " + cfg.resource_name().name +
+                                " as it hasn't been registered.");
+    }
+    try {
+        std::vector<std::string> implicit_deps = reg->validate(cfg);
+        for (int i = 0; i < implicit_deps.size(); i++) {
+            if (i == 0) {
+              response->add_dependencies();
```

Comment:
Oh wait, maybe my above comment is inadequate because of this case. I'm a little confused about what this `if i == 0` case is doing. Might be good to chat in person about it?

Reply:
Removed this logic; see my comment above.